src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -774,6 +774,203 @@ public static void MixedArrayIndexing()
+
+                Assert.True(root.TryGetDouble(out double doubleVal));
+                Assert.Equal(expectedDouble, doubleVal);
+
+                Assert.True(root.TryGetDecimal(out decimal decimalVal));
+                Assert.Equal(expectedDecimal, decimalVal);
+
+                Assert.True(root.TryGetSByte(out sbyte sbyteVal));
+                Assert.Equal(value, sbyteVal);
+
+                Assert.True(root.TryGetInt16(out short shortVal));
+                Assert.Equal(value, shortVal);
+
+                Assert.True(root.TryGetInt32(out int intVal));
+                Assert.Equal(value, intVal);
+
+                Assert.True(root.TryGetInt64(out long longVal));
+                Assert.Equal(value, longVal);
+
+                Assert.Equal(expectedFloat, root.GetSingle());
+                Assert.Equal(expectedDouble, root.GetDouble());
+                Assert.Equal(expectedDecimal, root.GetDecimal());
+                Assert.Equal(value, root.GetInt32());
+                Assert.Equal(value, root.GetInt64());
+
+                if (value >= 0)
+                {
+                    byte expectedByte = (byte)value;
+                    ushort expectedUShort = (ushort)value;
+                    uint expectedUInt = (uint)value;
+                    ulong expectedULong = (ulong)value;
+
+                    Assert.True(root.TryGetByte(out byte byteVal));
+                    Assert.Equal(expectedByte, byteVal);
+
+                    Assert.True(root.TryGetUInt16(out ushort ushortVal));
+                    Assert.Equal(expectedUShort, ushortVal);
+
+                    Assert.True(root.TryGetUInt32(out uint uintVal));
+                    Assert.Equal(expectedUInt, uintVal);
+
+                    Assert.True(root.TryGetUInt64(out ulong ulongVal));
+                    Assert.Equal(expectedULong, ulongVal);
+
+                    Assert.Equal(expectedUInt, root.GetUInt32());
+                    Assert.Equal(expectedULong, root.GetUInt64());
+                }
+                else
+                {
+                    Assert.False(root.TryGetByte(out byte byteValue));
+                    Assert.Equal(0, byteValue);
+
+                    Assert.False(root.TryGetUInt16(out ushort ushortValue));
+                    Assert.Equal(0, ushortValue);
+
+                    Assert.False(root.TryGetUInt32(out uint uintValue));
+                    Assert.Equal(0U, uintValue);
+
+                    Assert.False(root.TryGetUInt64(out ulong ulongValue));
+                    Assert.Equal(0UL, ulongValue);
+
+                    Assert.Throws<FormatException>(() => root.GetUInt32());
+                    Assert.Throws<FormatException>(() => root.GetUInt64());
+                }
+
+                Assert.Throws<InvalidOperationException>(() => root.GetString());
+                const string ThrowsAnyway = "throws-anyway";
+                Assert.Throws<InvalidOperationException>(() => root.ValueEquals(ThrowsAnyway));
```

Comment:
There are a number of tests in this file that check for the InvalidOperationException on GetInt32(); the new one-and-two-byte-integer methods should be added to those tests as well.

Reply:
Adding the same! Thanks!